src/main/java/online/talkandtravel/service/impl/AuthenticationServiceImpl.java

```diff
@@ -172,6 +173,14 @@ public AuthResponse login(String email, String password) {
     return createNewAuthResponse(jwtToken, authenticatedUser);
   }
 
+  private UserDetails getUserDetails(String token) {
+    Long userId = tokenService.extractId(token);
+    return userRepository
```

Comment:
make this ```userRepository
.findById(userId)
.map(CustomUserDetails::new)
.orElseThrow(() -> new UsernameNotFoundException("User not found"));```
as public method in UserService. And then invoke here `userService.yourNewMethod()`